controllers/trainingjob/trainingjob_controller_test.go

```diff
@@ -617,6 +617,301 @@ var _ = Describe("Reconciling a TrainingJob that exists", func() {
 				})
 			})
 		})
+
+		Context("TrainingJob is 'Failed' and has debug job with status 'InProgress'", func() {
+			var failureReason string
+			var debugJobStatus []sagemaker.RuleEvaluationStatus
+
+			BeforeEach(func() {
+				expectedStatus = sagemaker.TrainingJobStatusFailed
+				failureReason = "Failure within the training job"
+				debugJobStatus = append(debugJobStatus, sagemaker.RuleEvaluationStatusInProgress)
+				debugJobStatus = append(debugJobStatus, sagemaker.RuleEvaluationStatusIssuesFound)
+
+				// Add the failure reason to the describe output
+				describeOutput := CreateDescribeOutputWithDebugJobStatus(expectedStatus, expectedSecondaryStatus, debugJobStatus)
+				describeOutput.FailureReason = ToStringPtr(failureReason)
+
+				mockSageMakerClientBuilder.
+					AddDescribeTrainingJobResponse(describeOutput)
+			})
+
+			When("!HasDeletionTimestamp", func() {
+				It("Requeues after interval", func() {
+					ExpectRequeueAfterInterval(reconcileResult, reconcileError, pollDuration)
+				})
+
+				It("Updates status", func() {
+					ExpectAllJobsStatusToBe(trainingJob, string(expectedStatus), string(expectedSecondaryStatus), debugJobStatus)
```

Comment:
`ExpectAllJobsStatusToBe` name is confusing, can do either `ExpectStatusAndDebugStatusToBe` or separate `ExpectStatusToBe` and `ExpectDebugJobStatusesToBe`.

Reply:
++ on separate still